src/middleware/navigation/update/has.given.valid.beneficial.owner.details.middleware.ts

```diff
@@ -0,0 +1,18 @@
+import { Request, Response, NextFunction } from 'express';
+
+import { logger } from '../../../utils/logger';
+import { SECURE_UPDATE_FILTER_URL } from '../../../config';
+import { getApplicationData } from "../../../utils/application.data";
+import { checkHasGivenValidBOData, NavigationErrorMessage } from '../check.condition';
+
+export const hasGivenValidBODetails = (req: Request, res: Response, next: NextFunction): void => {
+  try {
+    if (!checkHasGivenValidBOData(getApplicationData(req.session), req.params['beneficialOwnerType'], req.params['id'])) {
+      logger.infoRequest(req, NavigationErrorMessage);
+      return res.redirect(SECURE_UPDATE_FILTER_URL);
```

Comment:
This redirect does not look right

Reply:
I do agree, and its something I've spoke with Matthew about before, but yeah .. all other default behaviour of nav checks is it redirects to 1st page of the journey if a nav check fails.

Reply:
Better to redirect to bo/mo type controller.

Reply:
Yeah definitely, I've updated that now